### PR TITLE
Feat/mongo

### PR DESCRIPTION
**Pull request by febinct**
added operator find & find_one for mongo

### CODE REVIEW

**Comment by Navaneeth-pk** [comment]
You might need to use the npm version `7.20.0` and run `npm install` again or revert the changes made to lock files as there aren't any changes to `package.json` files.
@febinct

File: frontend/src/Editor/QueryManager/QueryEditors/Mongodb.jsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import SelectSearch, { fuzzySearch } from 'react-select-search';
 import { CodeHinter } from '../../CodeBuilder/CodeHinter';
 import { placeholders } from './placeholders';
 import { changeOption } from './utils';
+import { NumberInput } from '../../Components/NumberInput';
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/QueryManager/QueryEditors/Mongodb.jsx:7]: Unused import, please remove.

File: frontend/src/Editor/Components/NumberInput.jsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ export const NumberInput = function NumberInput({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [newNumber]);
 
-  const placeholder = component.definition.properties.placeholder.value;
+  const placeholder = component.definition.properties?.placeholder?.value ?? 0;
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Components/NumberInput.jsx:28]: Is this change necessary?

File: server/plugins/datasources/mongodb/index.ts [hunk 3]
```diff
@@ -22,6 +22,23 @@ export default class MongodbService implements QueryService {
...
           break;
+        case 'find_one':
+          result = await db.collection(queryOptions.collection).findOne(JSON.parse(queryOptions['query']));
+          break;
+        case 'find':
+          let resultQueryBuilder = db.collection(queryOptions.collection)
+          if ("query" in queryOptions && queryOptions['query'])
+            resultQueryBuilder = resultQueryBuilder.find(JSON.parse(queryOptions['query']))
+          else
+            resultQueryBuilder = resultQueryBuilder.find({})
```
Comment (spmurrayzzz) [inline server/plugins/datasources/mongodb/index.ts:33]: _(Sorry for chiming in abruptly here, was working on a similar feature locally while I was evaluating tooljet for internal use at my org and came across this thread)_

Did this work for you locally on large find queries?

Pretty sure what you need here is `resultQueryBuilder.find({}).toArray()` given that the upstream service is [awaiting the return value](https://github.com/ToolJet/ToolJet/blob/6acc4b43ab663a885a2498e966074e37e39a4300/server/src/services/data_queries.service.ts#L81). `find()` itself just[ returns a cursor](https://mongodb.github.io/node-mongodb-native/4.0/classes/collection.html#find) so that upstream expression will yield no actual documents.
